src/main/java/org/wiremock/grpc/dsl/WireMockGrpcService.java

```diff
@@ -51,4 +53,23 @@ public GrpcVerification verify(int count, String method) {
   public GrpcVerification verify(CountMatchingStrategy countMatch, String method) {
     return new GrpcVerification(wireMock, countMatch, serviceName, method);
   }
+
+  /** Resets all active mocks for the current gRPC service */
+  public void resetAllForService() {
```

Comment:
Having "resetAll" in the name implies (based on what this means in WireMock core) that the request log and scenarios would be reset also.

Also, this method implementation removes stub mappings regardless of whether they were loaded from the filesystem, whereas the corresponding reset method preserves the stubs that were provided by the filesystem and removes only the transient ones.

Reply:
> Having "resetAll" in the name implies (based on what this means in WireMock core) that the request log and scenarios would be reset also.

I did not think of that. Would something like `resetMocksForService()` be better? Or more simply `resetMocks()`? I was trying to indicate in the name that the reset was specific just for this one particular service.

> Also, this method implementation removes stub mappings regardless of whether they were loaded from the filesystem, whereas the corresponding reset method preserves the stubs that were provided by the filesystem and removes only the transient ones.

That didn't seem to be the case when I tested in one of my applications. Stubs that were loaded from the filesystem remained.

I can try to update the test case here with a test that proves (or disproves) it. If you are indeed correct I will update accordingly so that it does not do that.

Reply:
Hi @tomakehurst I've made some changes based on your review comments. I've also updated the test which now proves that the reset preserves the stubs that were provided by the filesystem and removes only the transient ones.